source/engines/rollEngine.js

```diff
@@ -1,548 +1,771 @@
-	}
-}
-
-class ResultBundle {
-	#resultset
-	#extraText
-
-	constructor(rs, et = ""){
-		var resultSet;
-		if (rs instanceof ResultSet) { // If it is a result set, just use it as-is
-			resultSet = rs;
-		} else { // Parse the string into a roll set
-			resultSet = ResultSet.parse(rs);
-		}
-		this.#resultset = resultSet;
-		this.#extraText = et;
-	}
-
-	getResultSet() {
-		return this.#resultset;
-	}
-
-	getExtraText() {
-		return this.#extraText;
-	}
-
-	toString(frac = false, simple = false) {
-		if (simple && frac) {
-			return `${this.#resultset.getResults()}/${this.#resultset.getMaxResults()}${this.#extraText ? ` ${this.#extraText}` : ''}`;
-		} else if (simple && !frac) {
-			return `${this.#resultset.getResults()}${this.#extraText ? ` ${this.#extraText}` : ''}`;
-		} else if (!simple && frac) {
-			return `${this.#resultset.toString(frac)} = ${this.#resultset.getResults()}/${this.#resultset.getMaxResults()}${this.#extraText ? ` ${this.#extraText}` : ''}`;
-		} else {
-			return `${this.#resultset.toString(frac)} = ${this.#resultset.getResults()}${this.#extraText ? ` ${this.#extraText}` : ''}`;
-		}
-	}
-}
-
-function parseRoll(input) { //This needs some wild revision before it'll work
-	input = input.replace(/\(/g,' ( ').replace(/\)/g,' ) ')
-	var splitInput = input.split(' ');
-	var rollArray = splitInput.reduce((acc, cur) => {
-		if (acc[acc.length-1] == '║' || cur == '') {
-			return acc;
-		} else if (/[\+\-\*]/.test(cur) || /([0-9]+|[0-9]+d[0-9]+([dk][lh]?[0-9]+)?)([\+\-\*]([0-9]+|[0-9]+d[0-9]+([dk][lh]?[0-9]+)?))*$/.test(cur) || /\(/.test(cur) || /\)/.test(cur)) {
-			acc.push(cur);
-			return acc;
-		} else { //stop on the first non-roll
-			acc.push('║');
-			return acc;
-		}
-	}, []);
-	rollArray[rollArray.length-1] == '║' ? rollArray.pop() : ()=>{};
-	var textArray = splitInput.reduceRight((acc, cur) => {
-		if (acc[0] == '║' || cur == '') {
-			return acc;
-		} else if (/([0-9]+|[0-9]+d[0-9]+([dk][lh]?[0-9]+)?)([\+\-\*]([0-9]+|[0-9]+d[0-9]+([dk][lh]?[0-9]+)?))*$/.test(cur) || /[0-9]+$/.test(cur) || /\)/.test(cur)) {
-			acc.unshift('║');
-			return acc;
-		} else {
-			acc.unshift(cur);
-			return acc;
-		}
-	}, []);
-	textArray[0] == '║' ? textArray.shift() : ()=>{};
-	return new ResultBundle(rollArray.join(''),textArray.join(' '));
-}
-
-function getRollString(input, frac = false, simple = false) {
-	return parseRoll(input).toString(frac, simple);
-}
-
-module.exports = {parseRoll, getRollString}
+class Result {
+	#result;
+
+	constructor(result) {
+		this.#result = result;
+	}
+
+	getResult() {
+		return this.#result;
+	}
+
+	toString() {
+		return `${this.#result}`;
+	}
+}
+
+class RollResult extends Result {
+	#dieSides;
+
+	constructor(dieSides) {
+		super(Math.floor(Math.random() * dieSides) + 1);
+		this.#dieSides = dieSides;
+	}
+
+	getMaxValue() {
+		return this.#dieSides;
+	}
+
+	toString(frac = false) {
+		return frac ? `${this.getResult()}/${this.getMaxValue()}` : `${this.getResult()}`;

```

Comment:
```suggestion
		if (frac) {
			return `${this.getResult()}/${this.getMaxValue()}`;
		} else {
			return this.getResult().toString();
		}
```
Two thoughts here:
1. I treat ternaries as a "necessary evil".
2. `${this.getResult()}` is the same as `${this.toString()}` which is the same as `this.getResult().toString()`

Reply:
1. I typically prefer to use ternaries for simple switches, especially on string output, like here.
2. As much as in this one location this _appears_ to be the case. I'm very hesitant to change how the strings are constructed here. It started like this due to the potential need for quickly modifying how the string is constructed, and it ultimately follows the same templating for how the more complex `toString` functions are implemented later. Additionally, those more complex instances of the nodes in the parsed operation tree treat `getResult` and `toString` very differently, and, as much as `toString` is a default object operation, crossing those wires by using both names within the same function body here would likely lead to the potential for confusion.